src/ApplicationInsights.Kubernetes/TelemetryInitializers/SimplePerformanceCounterTelemetryInitializer.cs

```diff
@@ -0,0 +1,69 @@
+#if NETSTANDARD2_0
+
+using System;
+using System.Diagnostics;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using static Microsoft.ApplicationInsights.Kubernetes.TelemetryProperty;
+
+namespace Microsoft.ApplicationInsights.Kubernetes
+{
+    internal class SimplePerformanceCounterTelemetryInitializer : ITelemetryInitializer
+    {
+        TimeSpan _lastCPUSample;
+        Stopwatch _cpuWatch = new Stopwatch();
+
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (telemetry is ISupportMetrics metricsTelemetry)
+            {
+                // Write CPU metrics
+                TimeSpan endCPUTime = GetCPUTime();
+                if (_lastCPUSample != default)
+                {
+                    if (_cpuWatch.ElapsedMilliseconds > 0)
+                    {
+                        int processorCount = Environment.ProcessorCount;
+                        Debug.Assert(processorCount > 0, $"How could process count be {processorCount}?");
+                        // A very simple but not that accurate evaluation of how much CPU the process is take out of a core.
+                        double cpuPercentage = (endCPUTime - _lastCPUSample).TotalMilliseconds / (_cpuWatch.ElapsedMilliseconds);
+                        SetMetrics(metricsTelemetry, ProcessProperty.CPUPrecent, cpuPercentage);
+                    }
+                    else
+                    {
+                        _cpuWatch.Start();
+                    }
+                }
+
+                // Snap CPU usage
+                _lastCPUSample = endCPUTime;
+                _cpuWatch.Restart();
+
+                // Write memory metrics
+                long memory = GetMemory();
+                SetMetrics(metricsTelemetry, ProcessProperty.Memory, memory);
+            }
+        }
+
+        private void SetMetrics(ISupportMetrics telemetry, string key, double value)
+        {
+            if (!telemetry.Metrics.ContainsKey(key))
+            {
+                telemetry.Metrics.Add(key, value);
+            }
+        }
+
+        private TimeSpan GetCPUTime() => GetFromCurrentProcess<TimeSpan>(p => p.TotalProcessorTime);
```

Comment:
Creating and disposing of a `Process` object is somewhat expensive. Consider caching it.

Reply:
Done.